File: src/main/java/com/xeoneux/caravansary/entity/RoomEntity.java
package com.xeoneux.caravansary.entity;

import javax.persistence.*;
import javax.validation.constraints.NotNull;

@Entity
@Table(name = "Room")
public class RoomEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;

    @NotNull private Integer roomNumber;

    @NotNull private String price;

    public RoomEntity() {
        super();
    }

    public RoomEntity(Integer roomNumber, String price) {
        super();
        this.roomNumber = roomNumber;
        this.price = price;
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public Integer getRoomNumber() {
        return roomNumber;
    }

    public void setRoomNumber(Integer roomNumber) {
        this.roomNumber = roomNumber;
    }

    public String getPrice() {
        return price;
    }

    public void setPrice(String price) {
        this.price = price;
    }
}

File: src/main/kotlin/com/xeoneux/caravansary/entity/RoomEntity.kt
package com.xeoneux.caravansary.entity

import javax.persistence.*
import javax.validation.constraints.NotNull
import java.util.ArrayList

@Entity
@Table(name = "Room")
data class RoomEntity(@Id
                      @GeneratedValue(strategy = GenerationType.AUTO)
                      var id: Long? = null,

                      @NotNull
                      var roomNumber: Int? = null,

                      @NotNull
                      var price: String? = null,

                      @OneToMany(fetch = FetchType.EAGER, cascade = arrayOf(CascadeType.PERSIST))
                      private var reservationEntityList: MutableList<ReservationEntity>? = null) {

    constructor(roomNumber: Int, price: String) : this(null, roomNumber, price)

    fun addReservationEntity(reservationEntity: ReservationEntity) {
        if (reservationEntityList == null) reservationEntityList = ArrayList()

        reservationEntityList!!.add(reservationEntity)
    }
}
